test/e2e_node/memory_manager_test.go

```diff
@@ -802,7 +936,757 @@ var _ = SIGDescribe("Memory Manager Incompatibility Pod Level Resources", framew
 				initCtnParams = []memoryManagerCtnAttributes{}
 			})
 
-			ginkgo.It("should not report any memory data during request to pod resources List when pod has pod level resources", func(ctx context.Context) {
+			ginkgo.It("scope: pod, should allocate exclusive memory to a guaranteed pod with pod-level resources and guaranteed container, PodLevelResourceManagers enabled", func(ctx context.Context) {
+				currentCfg, err := getCurrentKubeletConfig(ctx)
+				framework.ExpectNoError(err)
+				updateKubeletConfigIfNeeded(ctx, f, configureMemoryManagerInKubelet(currentCfg, &memoryManagerKubeletArguments{
+					policyName:                     string(staticPolicy),
+					topologyManagerPolicy:          topologymanager.PolicyBestEffort,
+					topologyManagerScope:           topologymanager.PodTopologyScope,
+					enablePodLevelResources:        true,
+					enablePodLevelResourceManagers: true,
+				}))
+
+				ctnParams = []memoryManagerCtnAttributes{
+					{
+						ctnName: "gu-container",
+						cpus:    "100m",
+						memory:  "128Mi",
+					},
+				}
+				testPod = makeMemoryManagerPod("gu-pod-level-gu-ctn", nil, ctnParams)
+				testPod.Spec.Resources = &v1.ResourceRequirements{
+					Limits: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("100m"),
+						v1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("100m"),
+						v1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+				}
+
+				ginkgo.By("Running the test pod")
+				testPod = e2epod.NewPodClient(f).CreateSync(ctx, testPod)
+
+				if !*isMultiNUMASupported {
+					return
+				}
+
+				verifyMemoryPinning(f, ctx, testPod, []int{0})
+				verifyMemoryManagerAllocations(f, ctx, testPod, []string{
+					"gu-container",
+				})
+			})
+
+			ginkgo.It("scope: pod, should allocate exclusive memory to a guaranteed pod with pod-level resources and non-guaranteed containers, PodLevelResourceManagers enabled", func(ctx context.Context) {
+				currentCfg, err := getCurrentKubeletConfig(ctx)
+				framework.ExpectNoError(err)
+				updateKubeletConfigIfNeeded(ctx, f, configureMemoryManagerInKubelet(currentCfg, &memoryManagerKubeletArguments{
+					policyName:                     string(staticPolicy),
+					topologyManagerPolicy:          topologymanager.PolicyBestEffort,
+					topologyManagerScope:           topologymanager.PodTopologyScope,
+					enablePodLevelResources:        true,
+					enablePodLevelResourceManagers: true,
+				}))
+
+				testPod = &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "gu-pod-level-ngu-ctn-",
+					},
+					Spec: v1.PodSpec{
+						RestartPolicy: v1.RestartPolicyNever,
+						Containers: []v1.Container{
+							{
+								Name:    "ngu-container",
+								Image:   busyboxImage,
+								Command: []string{"sh", "-c", "grep Mems_allowed_list /proc/self/status | cut -f2 && sleep 1d"},
```

Comment:
Is better to check the container cgroup content rather than have a dependency on the container output. Can be fixed later.